dataland-frontend/src/components/resources/frameworkDataSearch/lksg/LksgCompanyDataTable.vue

```diff
@@ -41,28 +41,28 @@
         :key="reportingPeriod.dataId"
       >
         <template #body="{ data }">
-          <a
-            v-if="
-              Array.isArray(data[reportingPeriod.dataId]) &&
-              (data[reportingPeriod.dataId].length > 1 ||
-                data[reportingPeriod.dataId].some((el) => typeof el === 'object'))
-            "
-            @click="openModalAndDisplayValuesInSubTable(data[reportingPeriod.dataId], data.kpiLabel, data.kpiKey)"
-            class="link"
-            >Show "{{ data.kpiLabel }}"
-            <em class="material-icons" aria-hidden="true" title=""> dataset </em>
-          </a>
-          <span v-else-if="typeof data[reportingPeriod.dataId] === 'object' && data[reportingPeriod.dataId]?.value">
-            {{ data[reportingPeriod.dataId].value }}
-          </span>
-
-          <span v-else
-            >{{
-              Array.isArray(data[reportingPeriod.dataId])
-                ? data[reportingPeriod.dataId][0]
-                : data[reportingPeriod.dataId]
-            }}
-          </span>
+          <template v-if="data[reportingPeriod.dataId] !== undefined && data[reportingPeriod.dataId] !== null">
+            <template v-if="Array.isArray(data[reportingPeriod.dataId])">
+              <a
+                v-if="
+                  data[reportingPeriod.dataId].length > 1 ||
+                  data[reportingPeriod.dataId].some((el) => typeof el === 'object')
+                "
+                @click="openModalAndDisplayValuesInSubTable(data[reportingPeriod.dataId], data.kpiLabel, data.kpiKey)"
+                class="link"
+                >Show "{{ data.kpiLabel }}"
+                <em class="material-icons" aria-hidden="true" title=""> dataset </em>
+              </a>
+              <span v-else> data[reportingPeriod.dataId][0] </span>
+            </template>
+            <span v-else-if="data.kpiFormFieldComponent == 'PercentageFormField'">
+              {{ data[reportingPeriod.dataId] }} %</span
+            >
+            <span v-else-if="typeof data[reportingPeriod.dataId] === 'object' && data[reportingPeriod.dataId]?.value">
+              {{ data[reportingPeriod.dataId].value }}
```

Comment:
Why do you assume that an object could have a field `.value` ?   I mean where does this assumption come from and why is this explicitly handled here?

Reply:
Doesn't this mean, that all objects that we pass to an lksg dataset field, must have a field `value`, otherwise they will not be displayed at all?

Reply:
I think this is for filtering data points. They are of a type which has it's actual value stored in a field called value.

Reply:
Ok if that is the case then...

1.) The type check is not sufficient. It checks if the kpiValue is of type "object". But it is not guaranteed that each "object" type kpiValue is of type DataPoint.  

2.) The Lksg model does not even have a DataPoint-type field.   So it does not make sense in the first place.

My point is that I think this code is dead and instable, because it does an assumption (it assumes that all "object" type kpiValues have a field ".value")

Reply:
It is not assumed that all objects have type DataPoint. That is why the existence of the `value` field is checked.
Ad 2: Hmmm. Yes. Maybe my guess was wrong. However, this is how we used it in the document download branch.

Reply:
The second check is to ensure the object has a field `value`

Reply:
`data[reportingPeriod.dataId]?.value` is falsy if .value returns undefined

Reply:
Ah ok.  Then 1) is answered. Thanks. I overlook that.

But still this is dead code currently.   Should we therefore remove it?
What do you think?

Reply:
Well in sfdr `DataPointBigDecimal` and `DataPointYesNo` are actually being used.  Both are objects with a `value` field.
There the code would not be dead.

Maybe it was copy pasted?

Reply:
Florian and I decided to not remove this code. It is true that it is currently not executed, but it assures that actual DataPoint objects can be displayed, so it kind of makes this template more "stable" and "safe" for future changes of the data model.